src/readfish/plugins/dorado.py

```diff
@@ -0,0 +1,278 @@
+"""Dorado plugin module
+
+Extension of pyBaseCaller that maintains a connection to the basecaller
+"""
+
+from __future__ import annotations
+import logging
+import os
+import time
+from collections import namedtuple
+from pathlib import Path
+from typing import Iterable, TYPE_CHECKING
+
+import numpy as np
+import numpy.typing as npt
+from minknow_api.protocol_pb2 import ProtocolRunInfo
+
+try:
+    from pybasecall_client_lib.helper_functions import package_read
+    from pybasecall_client_lib.pyclient import PyBasecallClient
+except ImportError:
+    pass
+
+from readfish._loggers import setup_logger
+from readfish.plugins.abc import CallerABC
+from readfish.plugins.utils import Result
+from readfish._utils import nice_join
+
+
+if TYPE_CHECKING:
+    import minknow_api
+
+__all__ = ["Caller"]
+
+logger = logging.getLogger("RU_basecaller")
+CALIBRATION = namedtuple("calibration", "scaling offset")
+
+
+class DefaultDAQValues:
+    """Provides default calibration values
+
+    Mimics the read_until_api calibration dict value from
+    https://github.com/nanoporetech/read_until_api/blob/2319bbe/read_until/base.py#L34
+    all keys return scaling=1.0 and offset=0.0
+    """
+
+    calibration = CALIBRATION(1.0, 0.0)
+
+    def __getitem__(self, _):
+        return self.calibration
+
+
+_DefaultDAQValues = DefaultDAQValues()
+
+
+class Caller(CallerABC):
+    def __init__(
+        self, run_information: ProtocolRunInfo = None, debug_log=None, **kwargs
+    ):
```

Comment:
@Adoni5 is it possible to derive the sample rate from ProtocolRunInfo? We currently pass this in and if it contained the sample rate we would be golden?

Reply:
OK - so we need to pass this in then.

Reply:
No. Sample rate can be found in the [device proto](https://github.com/nanoporetech/minknow_api/blob/5.9.1/proto/minknow_api/device.proto#L241)

Reply:
yah - so we need to pass the device into the base caller. Should we handle it in the same way as run_information? So allow it to be None?